packages/eslint-plugin/src/rules/no-deprecated.ts

```diff
@@ -271,11 +271,32 @@ export default createRule({
       );
     }
 
+    function getJSXAttributeDeprecation(
+      openingElement: TSESTree.JSXOpeningElement,
+      propertyName: string,
+    ): string | undefined {
+      const tsNode = services.esTreeNodeToTSNodeMap.get(openingElement.name);
+
+      const contextualType = nullThrows(
+        checker.getContextualType(tsNode as ts.Expression),
+        'Expected jsx opening element name to have contextualType',
+      );
```

Comment:
[Testing] _Most_ of the time, when a `nullThrows` is added for a type retrieval, it's masking a situation where the type legitimately could be nullish. But I tried locally and couldn't break this one! 🤷 

In this case, if the contextual type can't be found, then TypeScript gives back the intrinsic error type: `{ flags: 1, intrinsicName: "error", ... }`. So that's fine.

To be thorough, could you please add some tests for this case?

* The namespace being `any` `unknown`, and instrinsic error
* The attribute being `any` `unknown`, and instrinsic error